src/main/java/com/yelp/nrtsearch/server/field/TextFieldDef.java

```diff
@@ -15,15 +15,71 @@
  */
 package com.yelp.nrtsearch.server.field;
 
+import com.yelp.nrtsearch.server.field.properties.PrefixQueryable;
 import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.grpc.IndexPrefixes;
+import com.yelp.nrtsearch.server.grpc.PrefixQuery;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.lucene.document.Document;
 import org.apache.lucene.document.FieldType;
 import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
 
 /** Field class for 'TEXT' field type. */
-public class TextFieldDef extends TextBaseFieldDef {
+public class TextFieldDef extends TextBaseFieldDef implements PrefixQueryable {
+  protected PrefixFieldDef prefixFieldDef;
+  private final Map<String, IndexableFieldDef<?>> childFieldsWithPrefix;
+  private static final int DEFAULT_MIN_CHARS = 2;
+  private static final int DEFAULT_MAX_CHARS = 5;
+
   public TextFieldDef(
       String name, Field requestField, FieldDefCreator.FieldDefCreatorContext context) {
     super(name, requestField, context);
+    if (requestField.hasIndexPrefixes()) {
+      verifySearchable("Prefix query");
+      int minChars =
+          requestField.getIndexPrefixes().hasMinChars()
+              ? requestField.getIndexPrefixes().getMinChars()
+              : DEFAULT_MIN_CHARS;
+      int maxChars =
+          requestField.getIndexPrefixes().hasMaxChars()
+              ? requestField.getIndexPrefixes().getMaxChars()
+              : DEFAULT_MAX_CHARS;
+      validatePrefix(minChars, maxChars);
+
+      this.prefixFieldDef =
+          new PrefixFieldDef(
+              getName(),
+              Field.newBuilder()
+                  .setSearch(true)
+                  .setAnalyzer(requestField.getAnalyzer())
+                  .setIndexAnalyzer(requestField.getIndexAnalyzer())
```

Comment:
What happens if the analyzers in the `requestField` are unset, will the new field values be counted as set? If so, it would need to use the `has` methods to only set the analyzers if they exist in the request.